protostar/commands/test/fuzzing/exceptions.py

```diff
@@ -1,5 +1,13 @@
 from protostar.protostar_exception import ProtostarException
+from protostar.commands.test.test_environment_exceptions import ReportedException
 
 
 class FuzzingError(ProtostarException):
     pass
+
+
+class HypothesisRejectException(ReportedException):
+    """
+    Hypothesis uses exceptions to handle `hypothesis.reject()` and `hypothesis.assume()`.
+    This class is used to smuggle them through the Cairo VM.
+    """
```

Comment:
Hypothesis may end up using `UnsatisfiedAssumption` to smuggle some information somewhen, so we should play it safe, and not reimplement their logic for rejections ourselves.

I think we should:
1. Store catched `UnsatisfiedAssumption` inside `HypothesisRejectException`
2. `throw HypothesisRejectException(unsatisifed_assumption_ex) from unsatisifed_assumption_ex` instead of `from None`
3. re-raise catched `unsatisifed_assumption_ex` in fuzzing loop